sdk/src/account.rs

```diff
@@ -21,24 +21,184 @@ pub struct Account {
     pub rent_epoch: Epoch,
 }
 
+/// An Account with data that is stored on chain
+/// This will become a new in-memory representation of the 'Account' struct data.
+/// The existing 'Account' structure cannot easily change due to downstream projects.
+/// This struct will shortly rely on something like the AnAccount trait for access to the fields.
+#[derive(Serialize, Deserialize, PartialEq, Eq, Clone, Default, AbiExample)]
+pub struct AccountNoData {
```

Comment:
I think `AccountRefData` or `AccountSharedData` might be better. `AccountNoData` sounds to me like there's no data at all, but I think it just means it has a reference to the data instead of owning it outright. Does it sound right?